onnxruntime/python/tools/microbench/fast_gelu.py

```diff
@@ -0,0 +1,55 @@
+import numpy as np
+from benchmark import benchmark 
+
+
+def create_inputs_outputs(batch, seq_len, intermediate_dimension, data_type):
+    np.random.seed(0)
+    a = np.random.rand(batch, seq_len, intermediate_dimension).astype(data_type)
+    b = np.random.rand(intermediate_dimension).astype(data_type)
+    c = np.random.rand(batch, seq_len, intermediate_dimension).astype(data_type)
+
+    inputs = {"A": a, "B": b}
+    outputs = {"return_val": c}
+ 
+    return inputs, outputs
+
+
+def add_benchmark_case(benchmark_cases, batch_size, seq_len, intermediate_dimension, data_type, model):
+    benchmark_cases += [
+        (batch_size, seq_len, intermediate_dimension, data_type, model),
+    ]
+
+
+def create_benchmark_cases(precision="fp16"):
+    benchmark_cases = []
+    if precision == "fp16":
+      model = "models/fast_gelu_fp16.onnx"
+      data_type = np.float16
+    else:
+      model = "models/fast_gelu_fp32.onnx"
+      data_type = np.float32
+
+    # bert-large
+    hidden_size = 1024
+    seq_len = 384
+    batch_size = 1
+    intermediate_dimension = hidden_size * 4
+    add_benchmark_case(benchmark_cases, batch_size, seq_len, intermediate_dimension, data_type, model)
+
+    return benchmark_cases
+
+
+def benchmark_fast_gelu(batch, seq_len, intermediate_dimension, data_type, onnx_file):
+    inputs, outputs = create_inputs_outputs(batch, seq_len, intermediate_dimension, data_type)
+    time = benchmark(onnx_file, inputs, outputs, "rocm")
```

Comment:
is this for rocm only? if so, should be move this under rocm dir? I accidentally run this on v100 and got EP error. 
another option is to take EP as param, then we can pass the provider down to benchmark. thought?
Same concern for matmul.py

Reply:
Good point! Added more user-friendly provider type assertion and also allow users to pass in provider type as an argument now.